### PR TITLE
Fix go mod

### PR DESCRIPTION
**Pull request by dshulyak**
go.mod directives were referencing wrong dependencies. I am not sure how they were generated, but all deps related to libp2p were referencing gx'ed versions (their own package manager). Maybe it was somehow converted from Gopkg.toml in status-go.

I converted several dependencies to go.mod (rendezvous and go-multiaddr-ethv4), and all other libp2p deps now will be imported correctly. So we can safely work with go modules (adding, updating, vendoring deps). 

Also i had to use modvendor from fork cause not-forked version doesn't handle replace directives when copying non-go dependencies.

fixes problem reported in https://github.com/status-im/status-console-client/pull/34#issuecomment-492070795

### CODE REVIEW

**Comment by dshulyak** [comment]
But everyone will have to use golang v1.12 now, since some indirect modules require it

File: Makefile [hunk 1]
```diff
@@ -40,10 +40,11 @@ install-linter:
 
 install-dev:
 	# a tool to vendor non-go files
-	go get -u github.com/goware/modvendor
+	# installed from a fork that doesn't ignore replace directive
+	GO111MODULE=auto go get -u github.com/dshulyak/modvendor
```
Comment (mandrigin) [inline Makefile]: hmm, maybe it should be a FIXME then?

Comment (dshulyak) [reply Makefile]: sure, will add FIXME

**Comment by adambabik** [comment]
> go.mod directives were referencing wrong dependencies. I am not sure how they were generated, but all deps related to libp2p were referencing gx'ed versions (their own package manager). Maybe it was somehow converted from Gopkg.toml in status-go.

They at some point removed old tags and converted them to gxed versions and started publishing new tags which could overlap with the old ones... eh.

**Pull request merged**